src/main/java/com/jcabi/http/request/FakeRequest.java

```diff
@@ -272,4 +294,50 @@ public FakeRequest withBody(final byte[] body) {
         );
     }
 
+    /**
+     * Make a similar request, which returns the provided body
+     *  if the url matching regexp.
+     * @param regexp Url regexp
+     * @param text Body
+     * @return New request
+     */
+    public FakeRequest withBody(final String regexp, final String text) {
+        return this.withBody(regexp, text.getBytes(FakeRequest.CHARSET));
+    }
+
+    /**
+     * Make a similar request, which returns the provided body
+     *  if the url matching regexp.
+     * @param regexp Url regexp
+     * @param body Body
+     * @return New request
+     */
+    public FakeRequest withBody(final String regexp, final byte[] body) {
+        final ConcurrentHashMap<Pattern, byte[]> map =
+            new ConcurrentHashMap<Pattern, byte[]>(this.content.size() + 1);
+        map.putAll(this.content);
+        map.put(Pattern.compile(regexp), body);
+        return new FakeRequest(
+            this.code,
+            this.phrase,
+            this.hdrs,
+            map
+        );
+    }
+
+    /**
+     * Looking for matching home.
+     * @param url Url
+     * @return Body
+     */
+    private byte[] bodyForHome(final String url) {
+        byte[] res = FakeRequest.EMPTY_BYTE_ARRAY;
+        for (final Map.Entry<Pattern, byte[]> entry : this.content.entrySet()) {
+            if (entry.getKey().matcher(url).matches()) {
+                res =  entry.getValue();
```

Comment:
@dmzaytsev thought so. @yegor256 what is the policy for this kind of situation? should we return from within the loop or break?